### PR TITLE
19 implement delete function

### PR DESCRIPTION
**Pull request by sixianli1995**

### CODE REVIEW

File: tests/test_routes.py [hunk 1]
```diff
@@ -173,13 +174,17 @@ def test_update_recommendation(self):
         self.assertEqual(updated_recommendation[PRODUCT_ID], 100)
         self.assertEqual(updated_recommendation[PRODUCT_NAME], "Hat")
 
-        response2 = self.client.put(
-            f"{BASE_URL}/99999",  #non-existing id
-            json=new_rec,
-            content_type=CONTENT_TYPE_JSON,
-        )
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
+    def test_delete_recommendation(self):
```
Comment (brady800526) [inline tests/test_routes.py:177]: Probably leave other's testing case still?

Comment (sixianli1995) [reply tests/test_routes.py:177]: Ok, I will fix it

File: tests/test_routes.py [hunk 2]
```diff
@@ -230,4 +242,14 @@ def test_create_recommendation_no_content_type(self):
...
+        """ It should Delete a Recommendation """
+        # create a recommendation to update
+        test_rec = self._create_recommendations(1)[0]
+        # delete the recommendation
 
+        response = self.client.delete(f"{BASE_URL}/{test_rec.id}")
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        # try to read the deleted recommendation
+        response = self.client.get(f"{BASE_URL}/{test_rec.id}")
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
```
Comment (brady800526) [inline tests/test_routes.py:255]: This is duplicate?

**Review by brady800526 — COMMENTED** [review]
Duplicate method

File: tests/test_routes.py (hunk 2 above)
Comment (sixianli1995) [reply tests/test_routes.py:255]: oh, sorry. This merge thing mess me up. I will delete this part.

**Review by brady800526 — COMMENTED** [review]
Nice

**Pull request merged**